src/Uno.UI/UI/Xaml/Controls/TreeView/TreeView.cs

```diff
@@ -0,0 +1,358 @@
+		private TreeViewList m_listControl;
+
+		public TreeView()
+		{
+			DefaultStyleKey = typeof(TreeView);
+
+			m_rootNode = new TreeViewNode();
+			m_pendingSelectedNodes = new List<TreeViewNode>();
+		}
+
+		internal TreeViewList ListControl { get => m_listControl; private set => m_listControl = value; }
+
+		public IList<TreeViewNode> RootNodes => m_rootNode.Children;
+
+		public object ItemFromContainer(DependencyObject container) =>
+			ListControl?.ItemFromContainer(container);
+
+		public DependencyObject ContainerFromItem(object item) =>
+			ListControl?.ContainerFromItem(item);
+
+		public TreeViewNode NodeFromContainer(DependencyObject container) =>
+			ListControl?.NodeFromContainer(container);
+
+		public DependencyObject ContainerFromNode(TreeViewNode node) =>
+			ListControl?.ContainerFromNode(node);
+
+		public TreeViewNode SelectedNode
+		{
+			get
+			{
+				return SelectedNodes.Count > 0 ? SelectedNodes[0] : null;
+			}
+			set
+			{
+				if (SelectedNodes.Count > 0)
+				{
+					SelectedNodes.Clear();
+				}
+				if (value != null)
+				{
+					SelectedNodes.Add(value);
+				}
+			}
+		}
+
+		public IList<TreeViewNode> SelectedNodes
+		{
+			get
+			{
+				return ListControl?.ListViewModel?.SelectedNodes ??
+					m_pendingSelectedNodes;
+				// we'll treat the pending selected nodes as SelectedNodes value
+				// if we don't have a list control or a view model
+			}
+		}
+
+		public object SelectedItem
+		{
+			get
+			{
+				return SelectedItems.Count > 0 ? SelectedItems[0] : null;
+			}
+			set
+			{
+				if (SelectedItems.Count > 0)
+				{
+					SelectedItems.Clear();
+				}
+				if (value != null)
+				{
+					SelectedItems.Add(value);
+				}
+			}
+		}
+
+		public IList<object> SelectedItems => ListControl?.ListViewModel?.SelectedItems;
+
+		internal void UpdateSelection(TreeViewNode node, bool isSelected)
+		{
+			var viewModel = ListControl?.ListViewModel;
+			if (viewModel != null)
+			{
+				if (isSelected != viewModel.IsNodeSelected(node))
+				{
+					var selectedNodes = viewModel.SelectedNodes;
+					if (isSelected)
+					{
+						if (SelectionMode == TreeViewSelectionMode.Single && selectedNodes.Count > 0)
+						{
+							selectedNodes.Clear();
+						}
+						selectedNodes.Append(node);
+					}
+					else
+					{
+						var index = selectedNodes.IndexOf(node);
+						if (index > -1)
+						{
+							selectedNodes.RemoveAt(index);
+						}
+					}
+				}
+			}
+		}
+
+		public void Expand(TreeViewNode node)
+		{
+			var vm = ListControl.ListViewModel;
+			vm.ExpandNode(node);
+		}
+
+		public void Collapse(TreeViewNode node)
+		{
+			var vm = ListControl.ListViewModel;
+			vm.CollapseNode(node);
+		}
+
+		public void SelectAll()
+		{
+			var vm = ListControl.ListViewModel;
+			vm.SelectAll();
+		}
+
+		private void OnItemClick(object sender, ItemClickEventArgs args)
+		{
+			var itemInvokedArgs = new TreeViewItemInvokedEventArgs(args.ClickedItem);
+			//TODO: m_itemInvokedEventSource(*this, *itemInvokedArgs);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)